eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -339,6 +342,10 @@ object Router {
     update_1_opt.foreach(u => assert(u.channelFlags.isNode1))
     update_2_opt.foreach(u => assert(!u.channelFlags.isNode1))
 
+    val nodeId1: PublicKey = ann.nodeId1
+    val nodeId2: PublicKey = ann.nodeId2
+    def shortChannelId: ShortChannelId = ann.shortChannelId
+    def channelId: ByteVector32 = toLongId(fundingTxid.reverse, outputIndex(ann.shortChannelId))
```

Comment:
That cannot work with dual-funding, where the `channelId` isn't derived from the funding tx anymore. I think we need to explicitly add it to the arguments of the `PublicChannel` class (it is derived from `open_channel2` and `accept_channel2`'s revocation basepoints).

Reply:
Since it's not _yet_ an issue, I suggest we fix it in #2224, because I'm changing the private->public channel logic.

Don't worry I'm keeping track of all the things that were deferred to #2224 😅

Reply:
Sounds good!